src/Azure/Orleans.Clustering.AzureStorage/OrleansSiloInstanceManager.cs

```diff
@@ -114,17 +114,12 @@ private static Uri ConvertToGatewayUri(SiloInstanceTableEntry gateway)
         public async Task<IList<Uri>> FindAllGatewayProxyEndpoints()
         {
             if (logger.IsEnabled(LogLevel.Debug)) logger.Debug(ErrorCode.Runtime_Error_100277, "Searching for active gateway silos for deployment {0}.", this.DeploymentId);
-            const string zeroPort = "0";
 
             try
             {
-                string filterOnPartitionKey = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.PartitionKey), QueryComparisons.Equal,
-                    this.DeploymentId);
-                string filterOnStatus = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.Status), QueryComparisons.Equal,
-                    INSTANCE_STATUS_ACTIVE);
-                string filterOnProxyPort = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.ProxyPort), QueryComparisons.NotEqual, zeroPort);
-                string query = TableQuery.CombineFilters(filterOnPartitionKey, TableOperators.And, TableQuery.CombineFilters(filterOnStatus, TableOperators.And, filterOnProxyPort));
-                var queryResults = await storage.ReadTableEntriesAndEtagsAsync(query);
+                const string ACTIVE = "Active";
+                const string ZERO= "0";
+                var queryResults = await storage.ReadTableEntriesAndEtagsAsync(TableOdataFilter.Create($"PartitionKey eq {DeploymentId} and Status eq {ACTIVE} and ProxyPort ne {ZERO}"));
```

Comment:
```suggestion
                var queryResults = await storage.ReadTableEntriesAndEtagsAsync(TableClient.CreateQueryFilter($"PartitionKey eq {DeploymentId} and Status eq {ACTIVE} and ProxyPort ne {ZERO}"));
```

Reply:
Looks like this should be updated to match the pattern on L216, as `TableOdataFilter` is `internal`.